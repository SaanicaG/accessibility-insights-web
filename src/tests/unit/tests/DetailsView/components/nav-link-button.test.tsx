// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { render } from '@testing-library/react';
import { onBaseLeftNavItemClick, onBaseLeftNavItemRender } from 'DetailsView/components/base-left-nav';
import { NavLinkButton, NavLinkButtonProps } from 'DetailsView/components/nav-link-button';
import * as React from 'react';
import { IMock, Mock } from 'typemoq';

describe('NavLinkButton', () => {
    let onClickNavLinkMock: IMock<onBaseLeftNavItemClick>;
    let onRenderNavLinkMock: IMock<onBaseLeftNavItemRender>;
    let props: NavLinkButtonProps;
    beforeEach(() => {
        onClickNavLinkMock = Mock.ofType<onBaseLeftNavItemClick>();
        onRenderNavLinkMock = Mock.ofType<onBaseLeftNavItemRender>();
        props = {
            link: {
                isExpanded: true,
                title: 'some title',
                onClickNavLink: onClickNavLinkMock.object,
                onRenderNavLink: onRenderNavLinkMock.object,
            },
            className: 'some class name',
        } as NavLinkButtonProps;
    });

    test('renders as button element', () => {
        const renderResult = render(<NavLinkButton {...props} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
    });

    test('renders with href set', () => {
        props.link.forceAnchor = true;
        const renderResult = render(<NavLinkButton {...props} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
    });

    test('renders with using name instead of title', () => {
        props.link.name = 'some name';
        const renderResult = render(<NavLinkButton {...props} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
    });

    test('renders with the proper aria-label', () => {
        props['aria-current'] = 'page';
        const renderResult = render(<NavLinkButton {...props} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
    });
});
