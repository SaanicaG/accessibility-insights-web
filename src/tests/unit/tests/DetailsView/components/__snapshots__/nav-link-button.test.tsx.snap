// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinkButton renders as button element 1`] = `
<DocumentFragment>
  <button
    aria-expanded="true"
    aria-label="some title"
    class="ms-Link navLinkButton some class name root-109"
    title="some title"
    type="button"
  />
</DocumentFragment>
`;

exports[`NavLinkButton renders with href set 1`] = `
<DocumentFragment>
  <a
    aria-expanded="true"
    aria-label="some title"
    class="ms-Link navLinkButton some class name root-110"
    href="#"
    title="some title"
  />
</DocumentFragment>
`;

exports[`NavLinkButton renders with the proper aria-label 1`] = `
<DocumentFragment>
  <button
    aria-expanded="true"
    aria-label="some title (current page)"
    class="ms-Link navLinkButton some class name root-109"
    title="some title"
    type="button"
  />
</DocumentFragment>
`;

exports[`NavLinkButton renders with using name instead of title 1`] = `
<DocumentFragment>
  <button
    aria-expanded="true"
    aria-label="some title"
    class="ms-Link navLinkButton some class name root-109"
    title="some title"
    type="button"
  />
</DocumentFragment>
`;
